File: app/src/main/java/ru/art2000/calculator/unit_converter/AllUnitsActivity.java
package ru.art2000.calculator.unit_converter;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import ru.art2000.calculator.R;
import ru.art2000.extensions.DayNightActivity;
import ru.art2000.helpers.PrefsHelper;

public class AllUnitsActivity extends DayNightActivity {

    double value;
    int pos;
    String[] dims;
    String category;
    Context mContext;

    @Override
    public boolean onSupportNavigateUp() {
        finish();
        return true;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        setTheme(PrefsHelper.getAppTheme());
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_all_units);
        mContext = this;
        setSupportActionBar(findViewById(R.id.toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        value = getIntent().getDoubleExtra("value", 1);
        pos = getIntent().getIntExtra("pos", 0);
        category = getIntent().getStringExtra("category");
        dims = getIntent().getStringArrayExtra("dims");
        String title = getResources().getStringArray(R.array.unit_converter_categories)
                [Formulas.getCategoryInt(category)];
        getSupportActionBar().setTitle(title);
        RecyclerView list = findViewById(R.id.all_units_list);
        list.addItemDecoration(new DividerItemDecoration(mContext, DividerItemDecoration.VERTICAL));
        LinearLayoutManager llm = new LinearLayoutManager(mContext);
        llm.setOrientation(RecyclerView.VERTICAL);
        list.setLayoutManager(llm);
        UnitListAdapter adapter = new UnitListAdapter(mContext, dims,
                Formulas.getCategoryInt(category), pos, value);
        list.setAdapter(adapter);
    }
}

File: unit/src/androidMain/kotlin/ru/art2000/calculator/unit/view/AllUnitsActivity.kt
package ru.art2000.calculator.unit.view

import android.os.Bundle
import android.view.View
import androidx.activity.viewModels
import dagger.hilt.android.AndroidEntryPoint
import dev.androidbroadcast.vbpd.viewBinding
import ru.art2000.calculator.common.view.AppActivity
import ru.art2000.calculator.unit.R
import ru.art2000.calculator.unit.databinding.ActivityAllUnitsBinding
import ru.art2000.calculator.unit.vm.UnitConverterModel
import ru.art2000.extensions.views.OrientationManger
import ru.art2000.extensions.views.addOrientationItemDecoration

@AndroidEntryPoint
internal class AllUnitsActivity : AppActivity(R.layout.activity_all_units) {

    private val model by viewModels<UnitConverterModel>()
    private val binding by viewBinding(ActivityAllUnitsBinding::bind)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val pos = intent.getIntExtra("highlightPosition", 0)

        setSupportActionBar(binding.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        val title =
            resources.getStringArray(R.array.unit_converter_categories)[model.category.ordinal]
        supportActionBar?.title = title

        val unitListAdapter = UnitListAdapter(
            this, this,
            model.converterFunctions, model.converterNames, model::copy, pos,
        )

        binding.allUnitsList.apply {
            layoutManager = OrientationManger(this@AllUnitsActivity)
            addOrientationItemDecoration()
            adapter = unitListAdapter
        }
    }

    override val topViews: List<View>
        get() = listOf(binding.root)

    override val bottomViews: List<View>
        get() = listOf(binding.allUnitsList)

    override val leftViews: List<View>
        get() = listOf(binding.root)

    override val rightViews: List<View>
        get() = listOf(binding.root)

    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }
}